### PR TITLE
Zscp: Full dynamic batch, remove batch size checks

### PR DESCRIPTION
**Pull request by kylesayrs**
This PR updates the zscp pipeline to account for new [dynamic batch](https://github.com/neuralmagic/deepsparse/pull/604). This includes
* Removing now unnecessarily error checks that constrain acceptable batch sizes
* Removing engine_forward logic that groups inferences by label
* Adding a check for divisible batch sizes (a little redundant given that the pipeline base class has a similar error, but this one makes it explicitly clear that the number of samples is the product of num_labels and num_sequences
* Adding tests to check correct pipeline instantiation

I've validated using the aformentioned tests as well as `eval_downstream` on sst2.

``` bash
deepsparse.transformers.eval_downstream zoo:nlp/text_classification/bert-base/pytorch/huggingface/mnli/12layer_pruned90-none -d sst2 --zero-shot True
```
sst2_zero_shot eval results: {'accuracy': 0.7798165137614679}
Expected: 0.779 (https://github.com/neuralmagic/deepsparse/pull/369)

### CODE REVIEW

**Comment by kylesayrs** [comment]
Not sure why this mock_engine test is failing. I wasn't able to reproduce the issue locally, and I haven't touched the test file where it fails.

**Comment by corey-nm** [comment]
> Not sure why this mock_engine test is failing. I wasn't able to reproduce the issue locally, and I haven't touched the test file where it fails. Will investigate further.

Yeah that's weird, I see a re-run even failed again. Just kicked off another re-run, probably just some weird anomaly. The timing tests are always fickle anyway 🤷

**Review by corey-nm — DISMISSED** [review]
A nice change :D love to see all these deletions

**Review by bfineran — APPROVED** [review]
LGTM

File: tests/deepsparse/transformers/pipelines/test_mnli_text_classification.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        pipeline = Pipeline.create(
+            "zero_shot_text_classification",
+            model_path=model_stub,
+            batch_size=batch_size,
+            labels=static_labels,
+        )
+
+    with inference_expectation:
+        pipeline(sequences=sequences, labels=labels)
+
```
Comment (bfineran) [inline tests/deepsparse/transformers/pipelines/test_mnli_text_classification.py:72]: should check result shape is correct here relative to batch size and num labels

**Review by bfineran — CHANGES_REQUESTED** [review]
quick comment on the test actually

File: tests/deepsparse/transformers/pipelines/test_mnli_text_classification.py (hunk 1 above)
Comment (kylesayrs) [reply tests/deepsparse/transformers/pipelines/test_mnli_text_classification.py:128]: Added a test case to test shapes

**Review by corey-nm — APPROVED** [review]

**Pull request merged**